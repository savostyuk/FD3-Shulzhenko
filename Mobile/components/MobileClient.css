tr.MobileClientInfo{
    border: 1px solid gray;
    border-collapse: collapse;
}
tr.MobileClientInfo td{
    padding: 5px 0;
}

td.statusactive{
    background-color: green;
}
td.statusblocked{
    background-color: red;
}