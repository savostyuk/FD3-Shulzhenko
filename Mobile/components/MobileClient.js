import React from 'react';
import PropTypes from 'prop-types';

import {clientEvents} from './events';

import './MobileClient.css';

class MobileClient extends React.PureComponent {

    static propTypes = {
        info:PropTypes.shape({
            id: PropTypes.number.isRequired,
            fam: PropTypes.string.isRequired,
            im: PropTypes.string.isRequired,
            otch: PropTypes.string.isRequired,
            balance: PropTypes.number.isRequired,
        }),
    };

    editClient = (EO) =>{
        clientEvents.emit('EEditClient', this.props.info.id);
    }

    deleteClient = (EO) =>{
        clientEvents.emit('EDeleteClient', this.props.info.id);
    }


    render() {

        console.log("MobileClient id="+this.props.info.id+" render");

        let status;
        if (this.props.info.balance>=0) status = 'active'
        else status = 'blocked';

        return (
                    <tr className='MobileClientInfo'>
                        <td>{this.props.info.fam}</td>
                        <td>{this.props.info.im}</td>
                        <td>{this.props.info.otch}</td>
                        <td>{this.props.info.balance}</td>
                        <td className={"status"+status}>{status}</td>
                        <td>
                            <input type='button' value='Редактировать'  onClick={this.editClient}/>
                        </td>
                        <td>
                            <input type='button' value='Удалить'  onClick={this.deleteClient}/>
                        </td>
                    </tr>

        );

    }

}

export default MobileClient;
