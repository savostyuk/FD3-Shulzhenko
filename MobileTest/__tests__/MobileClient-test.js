"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from "../components/MobileClient";

const client =   {id:101, fam:"Иванов", im:"Иван", otch:"Иванович", balance:200};

test ('Работа кнопки Удалить', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
        <MobileClient key={client.id} info={client} />
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // найдём в вёрстке компонента саму кнопку
    const buttonElem = component.root.find( el => el.props.className==='buttonDelete');
    // и "нажмём" на неё
    buttonElem.props.onClick();

    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // "нажмём" кнопку ещё раз
    buttonElem.props.onClick();

    // и получаем окончательный снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});