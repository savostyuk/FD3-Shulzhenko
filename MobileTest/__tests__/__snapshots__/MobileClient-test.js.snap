// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Работа кнопки Редактировать 1`] = `
<tr
  className="MobileClientInfo"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    className="statusactive"
  >
    active
  </td>
  <td>
    <input
      className="buttonEdit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      className="buttonDelete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`Работа кнопки Редактировать 2`] = `
<tr
  className="MobileClientInfo"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    className="statusactive"
  >
    active
  </td>
  <td>
    <input
      className="buttonEdit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      className="buttonDelete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`Работа кнопки Редактировать 3`] = `
<tr
  className="MobileClientInfo"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    className="statusactive"
  >
    active
  </td>
  <td>
    <input
      className="buttonEdit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      className="buttonDelete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`Работа кнопки Удалить 1`] = `
<tr
  className="MobileClientInfo"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    className="statusactive"
  >
    active
  </td>
  <td>
    <input
      className="buttonEdit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      className="buttonDelete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;

exports[`Работа кнопки Удалить 2`] = `
<tr
  className="MobileClientInfo"
>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    className="statusactive"
  >
    active
  </td>
  <td>
    <input
      className="buttonEdit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
    <input
      className="buttonDelete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
  </td>
</tr>
`;
